### DISCUSSION TITLE
Unable to Secure CloudStack System VMs with Dynamic URL

### DISCUSSION THREAD

**Discussion by vishnuvs369** [apache/cloudstack#9278] [Q&A] [answered]
##### ISSUE TYPE
* Other

##### COMPONENT NAME
`System VM's
`

##### CLOUDSTACK VERSION
`4.19.0.1
`

##### SUMMARY
I'm facing difficulties securing CloudStack System VMs using SSL configuration, as described in the documentation provided by ShapeBlue ( https://www.shapeblue.com/securing-cloudstack-4-11-with-https-tls/) 
Steps Taken:
1) Followed the steps outlined in the documentation to configure Dynamic URL for CloudStack System VMs.
2) Configured the required global settings as specified in the documentation.
3) Uploaded SSL certificates through the CloudStack UI for the system VMs.
4) Restarted the CloudStack management server and system VMs to apply the configuration changes.

Observations:
1) After completing the configuration steps, HTTPS access to the CloudStack System VMs is not functioning as expected.
2) Attempts to access the system VMs via HTTPS result in connection errors or unsuccessful HTTPS loading.
3) No visible indication of SSL encryption or secure connection is observed in the browser when accessing the System VMs.

![image](https://github.com/apache/cloudstack/assets/54799446/c18fbae4-d780-4316-b601-770ca6600b79)

**Comment by weizhouapache** [comment, marked as answer]
@vishnuvs369 
what are the related globals settings ?
check https://www.shapeblue.com/securing-cloudstack-4-11-with-https-tls/

**Comment by vishnuvs369** [comment]
@weizhouapache 
I enabled the following settings in global settings 
consoleproxy.ssl  True
secstorage.encrypt.copy True

Also Added domains in the below fields
consoleproxy.url.domain 
secstorage.ssl.cert.domain

**Comment by weizhouapache** [comment]
> @weizhouapache I enabled the following settings in global settings consoleproxy.ssl True secstorage.encrypt.copy True
> 
> Also Added domains in the below fields consoleproxy.url.domain secstorage.ssl.cert.domain

If you have restarted the management server, destroy the system vms and retry

**Comment by vishnuvs369** [comment]
@weizhouapache 
I restarted management server and destoyed system vm's, still its not working.

**Comment by weizhouapache** [comment]
> @weizhouapache I restarted management server and destoyed system vm's, still its not working.

@vishnuvs369 
is the DNS name resolvable ?
check if ip and port 443/8443 are reachable from the client/browser
check if port 443/8443 is on listen in CPVM
...

**Comment by vishnuvs369** [comment]
@weizhouapache 
Ports 443 and 8443 are listening in CPVM...and IP are reachable
![image](https://github.com/apache/cloudstack/assets/54799446/70144f4d-2818-42d5-b297-d767d79f56bc)

**Comment by weizhouapache** [comment]
> @weizhouapache Ports 443 and 8443 are listening in CPVM...and IP are reachable ![image](https://private-user-images.githubusercontent.com/54799446/324764500-70144f4d-2818-42d5-b297-d767d79f56bc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM4NjM3MzgsIm5iZiI6MTcxMzg2MzQzOCwicGF0aCI6Ii81NDc5OTQ0Ni8zMjQ3NjQ1MDAtNzAxNDRmNGQtMjgxOC00MmQ1LWIyOTctZDc2N2Q3OWY1NmJjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDIzVDA5MTAzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNhODY4MTk5MTM4MzNmMDU0MWRkNmEwN2IwYmM0Y2FlYjViYTg4ZWQ1ZGQ1Mjk4YTVmN2I0N2ZkMjg3ODA1NzkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.P2cfhADI-NUpCAN60iLCnS-dtWTsoYiIRYNYfO6p8-o)

any error/exception in /var/log/cloud.log in CPVM ?

**Comment by vishnuvs369** [comment]
@weizhouapache 
No errors found on CPVM..
![image](https://github.com/apache/cloudstack/assets/54799446/acecd838-7019-4474-97fb-a7a9975e2b19)

**Comment by weizhouapache** [comment]
> @weizhouapache No errors found on CPVM.. ![image](https://private-user-images.githubusercontent.com/54799446/324770823-acecd838-7019-4474-97fb-a7a9975e2b19.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM4NjQ0ODMsIm5iZiI6MTcxMzg2NDE4MywicGF0aCI6Ii81NDc5OTQ0Ni8zMjQ3NzA4MjMtYWNlY2Q4MzgtNzAxOS00NDc0LTk3ZmItYTdhOTk3NWUyYjE5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDIzVDA5MjMwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTlmY2E3YjBhNjdiNjg4NWIxNDQ5ZWQ2ZjlkNGJmYjg1ZDJmMzZiOWEyNzliNjc4NjBkZDJiYWNhYTcxYjQwMDYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.waYeEXdj3n_7GbdRexe_k0uRlMUJwSyrDe_AMMkKI5I)

is dns working ?

check dnsname:443 or dnsname:8443 by nc or telnet

**Comment by vishnuvs369** [comment]
@weizhouapache 
I tried a telnet to 443 and 8443....443 is not connected
![image](https://github.com/apache/cloudstack/assets/54799446/26027310-5031-4405-9aea-b21db0d3ac9c)

**Comment by weizhouapache** [comment]
> @weizhouapache I tried a telnet to 443 and 8443....443 is not connected ![image](https://private-user-images.githubusercontent.com/54799446/324776677-26027310-5031-4405-9aea-b21db0d3ac9c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM4NjUwNzUsIm5iZiI6MTcxMzg2NDc3NSwicGF0aCI6Ii81NDc5OTQ0Ni8zMjQ3NzY2NzctMjYwMjczMTAtNTAzMS00NDA1LTlhZWEtYjIxZGIwZDNhYzljLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDIzVDA5MzI1NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTMwZDQwN2NiZTIyYzkyMjZjNzJkMjQwYzUwYWNmMDA2N2VhMzliMmM3NzM2NGI4NTA5MDg0MWRiMWNlZTE0ZWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Zm1LfEnhFrbvKKn_UoZedgUsPAf_RHER0ACNPNDdWZI)

both 443/8443 should work

you can go to CPVM, and check if 127.0.0.1 443/8443 and public ip 443/8443 work

**Comment by vishnuvs369** [comment]
@weizhouapache 
I'm able to telnet 443/8443 from public IP and 127.0.0.1 but not able to telnet 443 from domain name where as 8443 is working.
where i have gone wrong?

![image](https://github.com/apache/cloudstack/assets/54799446/077d88d5-e5ec-4451-bfdd-7cf52aec040b)

**Comment by DaanHoogland** [comment]
> @weizhouapache I tried a telnet to 443 and 8443....443 is not connected ![image](https://private-user-images.githubusercontent.com/54799446/324776677-26027310-5031-4405-9aea-b21db0d3ac9c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM5NTE3NDgsIm5iZiI6MTcxMzk1MTQ0OCwicGF0aCI6Ii81NDc5OTQ0Ni8zMjQ3NzY2NzctMjYwMjczMTAtNTAzMS00NDA1LTlhZWEtYjIxZGIwZDNhYzljLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDI0VDA5MzcyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThkZmJiYTE2OWVkY2VkYjA3ZTYzYzIyM2I2MmJkZGEyNDY1ZGFhYjNmZDNlMmRlMDA5NjBjMGViZDkwZjUwMWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.m0BUJXllwjU2yz2zlXgMPKX09zNsZ0LeURxUzQwRW0Y)

@vishnuvs369 are these attempts from outside your cloud?
if yes, check if there is a firewall keeping you from connecting.
if no, can you share the firewall rules in your cpvm, and a traceroute from where you are trying to access your console?

**Comment by sureshanaparti** [comment]
> > @weizhouapache I tried a telnet to 443 and 8443....443 is not connected ![image](https://private-user-images.githubusercontent.com/54799446/324776677-26027310-5031-4405-9aea-b21db0d3ac9c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM5NTE3NDgsIm5iZiI6MTcxMzk1MTQ0OCwicGF0aCI6Ii81NDc5OTQ0Ni8zMjQ3NzY2NzctMjYwMjczMTAtNTAzMS00NDA1LTlhZWEtYjIxZGIwZDNhYzljLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDI0VDA5MzcyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThkZmJiYTE2OWVkY2VkYjA3ZTYzYzIyM2I2MmJkZGEyNDY1ZGFhYjNmZDNlMmRlMDA5NjBjMGViZDkwZjUwMWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.m0BUJXllwjU2yz2zlXgMPKX09zNsZ0LeURxUzQwRW0Y)
> 
> @vishnuvs369 are these attempts from outside your cloud? if yes, check if there is a firewall keeping you from connecting. if no, can you share the firewall rules in your cpvm, and a traceroute from where you are trying to access your console?

Hi @vishnuvs369 Any update on this, are you able to check the firewall rules, anything blocking there?
